### PR TITLE
Enable Mockito.DEEP_STUBS

### PR DESCRIPTION
**Pull request by jtama**
Fixes https://github.com/quarkiverse/quarkus-github-app/issues/153

### CODE REVIEW

File: testing/src/main/java/io/quarkiverse/githubapp/testing/internal/CallRealMethodAndSpyGHObjectResults.java [hunk 1]
```diff
@@ -20,6 +20,7 @@ public CallRealMethodAndSpyGHObjectResults(GitHubMockContextImpl mocks) {
     @Override
     public Object answer(InvocationOnMock invocation) throws Throwable {
         Object original = invocation.callRealMethod();
+        String packageName = original.getClass().getPackage().getName();
```
Comment (gsmet) [inline testing/src/main/java/io/quarkiverse/githubapp/testing/internal/CallRealMethodAndSpyGHObjectResults.java:23]: This looks like a useless change?

**Comment by gsmet** [comment]
@yrodiere could you have a look when you have some time?

File: testing/src/main/java/io/quarkiverse/githubapp/testing/internal/CallRealMethodAndSpyGHObjectResults.java (hunk 1 above)
Comment (jtama) [reply testing/src/main/java/io/quarkiverse/githubapp/testing/internal/CallRealMethodAndSpyGHObjectResults.java:23]: It definitely is. Some remaining debug code. Will delete it.

**Review by yrodiere — COMMENTED** [review]
I only have a few comments about code style, to simplify the code.

The changes seem fine, however:

1. Be warned that this may not work for methods starting with `get`, as there is some (complicated) tricks to make sure that we default to calling the real method for getters.
2. Because of the above, and also to prevent regressions, it would be great to have a simple test in `integration-tests/testing-framework/src/test/java/io/quarkiverse/githubapp/it/testingframework/TestingFrameworkTest.java` to demonstrate in which case this can be useful.

**Comment by jtama** [comment]
> Because of the above, and also to prevent regressions, it would be great to have a simple test in integration-tests/testing-framework/src/test/java/io/quarkiverse/githubapp/it/testingframework/TestingFrameworkTest.java to demonstrate in which case this can be useful.

Ok I will add some.

**Comment by jtama** [comment]
@yrodiere 
Hi,
I have been trying to implement tests for the deep-stub feature.

But for that I thought it would be cleaner to create a GitHubAppDeepStubTestingResource. The issue is that both are getting picked up when launching the tests as they both implement QuarkusTestResourceLifecycleManager.  So the newly added tests work, but the already there about missing mocks fails.

Would you have an hint ?

**Comment by yrodiere** [comment]
Introducing a new `GitHubAppDeepStubTestingResource` does not seem very future-proof anyway, as we may need to introduce other settings later.

Maybe a more appropriate way to expose the feature would be to introduce a dedicated annotation, like this:

```
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@QuarkusTestResource(GitHubAppTestingResource.class)
@Stereotype
public @interface GitHubAppTest {

    org.mockito.Answers defaultAnswer() default org.mockito.Answers.RETURNS_DEFAULTS;

}
```

Then you use it to enable GitHub app testing instead of `@QuarkusTestResource`:


```
@QuarkusTest
@GitHubAppTest(defaultAnswer = Answers.RETURNS_DEEP_STUBS)
public class MyTest {
```

And you can inspect the annotation and its attributes in `io.quarkiverse.githubapp.testing.internal.GitHubAppTestingCallback#afterConstruct`:

```
    @Override
    public void afterConstruct(Object testInstance) {
        if (!isEnabled()) {
            GitHubAppTestingContext.reset();
            return;
        }
        Answers defaultAnswer = Answers.RETURNS_DEFAULTS;
        GitHubAppTest annotation = testInstance.getClass().getAnnotation(GitHubAppTest.class);
        if ( annotation != null ) {
            defaultAnswer = annotation.defaultAnswer();
        }
        GitHubAppTestingContext.set(testInstance, defaultAnswer);
        GitHubAppTestingContext context = GitHubAppTestingContext.get();
        GitHubMockContextImpl mocks = context.mocks;
        QuarkusMock.installMockForType(mocks.service, GitHubService.class);
        QuarkusMock.installMockForType(mocks.fileDownloader, GitHubFileDownloader.class);
        QuarkusMock.installMockForType(context.errorHandler, ErrorHandler.class);
    }
```

If that doesn't work, it's probably due to classloader issues... let me know if I can help.

**Comment by jtama** [comment]
Yes much cleaner indeed !

File: docs/modules/ROOT/pages/includes/attributes.adoc [hunk 2]
```diff
@@ -1,4 +1,4 @@
-:quarkus-version: 1.13.3.Final
+:quarkus-version: 1.13.4.Final
```
Comment (yrodiere) [inline docs/modules/ROOT/pages/includes/attributes.adoc:1]: Seems unrelated? We're not releasing here.

File: testing/src/main/java/io/quarkiverse/githubapp/testing/internal/GitHubAppTestingCallback.java [hunk 3]
```diff
@@ -12,6 +12,7 @@
         implements QuarkusTestAfterConstructCallback {
 
     private static final String ENABLED_KEY = "quarkiverse-github-app-testing.enabled";
+    private static final String DEEP_STUBS_KEY = "quarkiverse-github-app-testing.deep-stubs.enabled";
```
Comment (yrodiere) [inline testing/src/main/java/io/quarkiverse/githubapp/testing/internal/GitHubAppTestingCallback.java:15]: I believe this is no longer necessary.

**Review by yrodiere — COMMENTED** [review]
There's a bit of cleanup left, but apart from that it looks good!

File: docs/modules/ROOT/pages/includes/attributes.adoc (hunk 2 above)
Comment (jtama) [reply docs/modules/ROOT/pages/includes/attributes.adoc:1]: Yep, 'mvn  clean package' changed the value, and I didn't cleaned it.

**Comment by yrodiere** [comment]
There was still an unnecessary change in `docs/modules/ROOT/pages/includes/attributes.adoc`, I don't know why.

I rebased on `main`, removed the change and force-pushed to your branch.

Will merge as soon as CI passes.

Thanks @jtama!

**Review by yrodiere — APPROVED** [review]

**Comment by jtama** [comment]
Thanks to you @yrodiere
